source/initial_conditions/adiabatic_boundary.cc

```diff
@@ -0,0 +1,277 @@
+ * and Wolfgang Bangerth.
+ */
+
+
+namespace aspect
+{
+  namespace InitialConditions
+  {
+    using namespace dealii;
+
+    /**
+     * Function that returns latitude and longitude from ECEF Cartesian
+     * coordinates that account for ellipsoidal shape of the Earth
+     * with WGS84 parameters.
+     */
+
+    template <int dim>
+    std::pair<double, double>
+    AdiabaticBoundary<dim>::lat_long_from_xyz_WGS84(const Point<3> &pos) const
+    {
+      /* Define WGS84 ellipsoid constants. */
+      const double radius = 6378137;
+      const double ellipticity = 8.1819190842622e-2;
+
+      const double b = std::sqrt(radius * radius
+                                 * (1 - ellipticity * ellipticity));
+      const double ep = std::sqrt((radius * radius - b * b) / (b * b));
+      const double p = std::sqrt(pos(0) * pos(0) + pos(1) * pos(1));
+      const double th = std::atan2(radius * pos(2), b * p);
+      const double lon = std::atan2(pos(1), pos(0));
+      const double lat = std::atan2((pos(2) + ep * ep * b * std::sin(th)
+                                     * std::sin(th) * std::sin(th)),
+                                    (p - (ellipticity
+                                          * ellipticity
+                                          * radius
+                                          * (std::cos(th) * std::cos(th)
+                                             * std::cos(th)))));
+
+      /* Convert to degrees. */
+      const double lon_degrees = lon * (180 / numbers::PI);
+      const double lat_degrees = lat * (180 / numbers::PI);
+
+      /* Set all longitudes between [0,360]. */
+      if (lon_degrees < 0)
+        return std::make_pair(lat_degrees, lon_degrees + 360);
+      else if (lon_degrees > 360)
+        return std::make_pair(lat_degrees, lon_degrees - 360);
+      else
+        return std::make_pair(lat_degrees, lon_degrees);
+
+    }
+
+    template <>
+    double
+    AdiabaticBoundary<2>::get_isotherm_depth(const double,
+                                             const double) const
+    {
+      AssertThrow(false, ExcInternalError());
+      return 0;
+    }
+
+    template <int dim>
+    double
+    AdiabaticBoundary<dim>::get_isotherm_depth(const double latitude,
+                                               const double longitude) const
+    {
+      /**
+       * Loop over the entire array and see if we find a point
+       * that's within delta of what we're looking for. If this
+       * point is not within the delta then test if it is within twice
+       * as much as delta.
+       */
+      for (unsigned int i = 0; i <= depths_iso.size();)
+        if (std::fabs(latitude - latitudes_iso[i]) <= delta && std::fabs(longitude - longitudes_iso[i]) < delta)
+          {
+            return -depths_iso[i]*1000;
+          }
+        else if (std::fabs(latitude - latitudes_iso[i]) <= delta*2 && std::fabs(longitude - longitudes_iso[i]) < delta*2)
```

Comment:
What is different if the distance is smaller than 2*delta but larger than delta? If you do the same thing in both cases this whole logic is the same as if you would only ask for `if distance < 2 * delta`.
Is there a meaning in the fact that you ask for `<=` in the first if, but `<` in the second?
And please add some parentheses around the two arguments of the `&&`. It might be correct as it is, but I am too lazy to remember the operator precedence and prefer to just use parentheses every time to avoid surprises.